lib/setup_functions.php

```diff
@@ -0,0 +1,965 @@
+        } else {
+            warn('external UK postcode table not found.');
+        }
+
+        if (CONST_Use_Extra_US_Postcodes) {
+            pgsqlRunScriptFile($aCMDResult, CONST_BasePath.'/data/us_postcode.sql');
+        }
+
+        if ($aCMDResult['no-partitions']) {
+            pgsqlRunScript($aCMDResult, 'update country_name set partition = 0');
+        }
+
+        // the following will be needed by create_functions later but
+        // is only defined in the subsequently called T
+        // Create dummies here that will be overwritten by the proper
+        // versions in create-tables.
+        pgsqlRunScript($aCMDResult, 'CREATE TABLE IF NOT EXISTS place_boundingbox ()');
+        pgsqlRunScript($aCMDResult, 'CREATE TYPE wikipedia_article_match AS ()', false);
+    }
+    return $aDSNInfo;
+}
+
+function importData($aCMDResult, $iCacheMemory, $aDSNInfo)
+{
+    info('Import data');
+
+    $osm2pgsql = CONST_Osm2pgsql_Binary;
+    if (!file_exists($osm2pgsql)) {
+        echo "Check CONST_Osm2pgsql_Binary in your local settings file.\n";
+        echo "Normally you should not need to set this manually.\n";
+        fail("osm2pgsql not found in '$osm2pgsql'");
+    }
+
+    if (!is_null(CONST_Osm2pgsql_Flatnode_File) && CONST_Osm2pgsql_Flatnode_File) {
+        $osm2pgsql .= ' --flat-nodes '.CONST_Osm2pgsql_Flatnode_File;
+    }
+    if (CONST_Tablespace_Osm2pgsql_Data)
+        $osm2pgsql .= ' --tablespace-slim-data '.CONST_Tablespace_Osm2pgsql_Data;
+    if (CONST_Tablespace_Osm2pgsql_Index)
+        $osm2pgsql .= ' --tablespace-slim-index '.CONST_Tablespace_Osm2pgsql_Index;
+    if (CONST_Tablespace_Place_Data)
+        $osm2pgsql .= ' --tablespace-main-data '.CONST_Tablespace_Place_Data;
+    if (CONST_Tablespace_Place_Index)
+        $osm2pgsql .= ' --tablespace-main-index '.CONST_Tablespace_Place_Index;
+    $osm2pgsql .= ' -lsc -O gazetteer --hstore --number-processes 1';
+    $osm2pgsql .= ' -C '.$iCacheMemory;
+    $osm2pgsql .= ' -P '.$aDSNInfo['port'];
+    if (isset($aDSNInfo['username']) && $aDSNInfo['username']) {
+        $osm2pgsql .= ' -U ' . $aDSNInfo['username'];
+    }
+    if (isset($aDSNInfo['hostspec']) && $aDSNInfo['hostspec']) {
+        $osm2pgsql .= ' -H ' . $aDSNInfo['hostspec'];
+    }
+
+    $aProcEnv = null;
+    if (isset($aDSNInfo['password']) && $aDSNInfo['password']) {
+        $aProcEnv = array_merge(array('PGPASSWORD' => $aDSNInfo['password']), $_ENV);
+    }
+
+    $osm2pgsql .= ' -d '.$aDSNInfo['database'].' '.$aCMDResult['osm-file'];
+    runWithEnv($osm2pgsql, $aProcEnv);
+
+    $oDB =& getDB();
+    if (!$aCMDResult['ignore-errors'] && !chksql($oDB->getRow('select * from place limit 1'))) {
+        fail('No Data');
+    }
+}
+
+function createFunctions($aCMDResult)
+{
+    info('Create Functions');
+
+    if (!checkModulePresence()) {
+        fail('error loading nominatim.so module');
+    }
+
+    createSqlFunctions($aCMDResult);
+}
```

Comment:
I'd leave that in the part that controls the flow of the setup and have onlc createSqlFunctions() and checkModulePresence() here.

Reply:
I am not sure I get what you mean with "that". I can move the info(.....) to the part that controls the flow. Then I would do it in all other functions as well. Or I could move the if to the checkModulePresenceparent function i.e. 
if !checkModulePresence()
   raise error
else
   createSqlFunctions

Reply:
Yes, I meant the whole thing with checkModulePresence() and createSqlFunctions(). It basically means removing this createFunctions() completely from this file.